Dockerfile

```diff
@@ -22,3 +22,19 @@ USER app
 COPY --chown=app:app ./ $APP_DIR
 
 CMD go run main.go
+
+
+FROM node:14.16-alpine3.13 AS web
```

Comment:
別ファイルにしていたweb用のDockerfileを削除し、ルートにあるDockerfileをマルチステージビルドに修正しました。

Reply:
ありがとうございます、、追加のerrrorが発生したため調査中です 🙇🏻 (docker-versionの問題だと思いますが)
```
yokoteru@yokosemominoMBP paddle % docker-compose up
ERROR: The Compose file './docker-compose.yml' is invalid because:
services.go.build contains unsupported option: 'target'
services.web.build contains unsupported option: 'target'
yokoteru@yokosemominoMBP paddle % git log
commit b06d14e19a74d9120ae491dd8613b0e1a5a048e3 (HEAD -> 3820/ch80/web-view-ui, origin/3820/ch80/web-view-ui)
Author: Hiroki Mori <xxx.bvb.taylor@gmail.com>
Date:   Sun Apr 11 23:32:48 2021 +0900

    docker multi-stage buildに変更
```

Reply:
docker-composeのバージョンが古い？
https://matsuand.github.io/docs.docker.jp.onthefly/compose/compose-file/compose-versioning/#version-23

Reply:
`docker-compose -v`？

Reply:
これですな。。
`yokoteru@yokosemominoMBP paddle % docker-compose -v
docker-compose version 1.23.2, build 1110ad01`

Reply:
要件を満たしているように見える…
> An upgrade of version 2.2 that introduces new parameters only available with Docker Engine version 17.06.0+. Version 2.3 files are supported by Compose 1.16.0+.

Reply:
1.29.0で解消しました　🙇🏻 
お手数おかけしました　🙏🏻

Reply:
これかな？
https://github.com/docker/compose/issues/5714

Reply:
酷くね？`3.8`までymlバージョンを上げます。「2019-07-22」以降のDockerを使用していれば問題なく動きます。
> Also pretty confusing for me, since this is documented in version 2 as well. I assumed that if it was available in 2.3 then naturally it'd also automatically be available in 3. Turns out that's not the case, for some reason.

Reply:
このエラーはまだ出てますか？
```
error Could not write file "/home/app/yarn-error.log": "EACCES: permission denied, open '/home/app/yarn-error.log'"
error An unexpected error occurred: "EACCES: permission denied, mkdir '/home/app/node_modules'".
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
paddle_web_1 exited with code 1
```

Reply:
そうですね、、

1. [ホスト側のdirの権限　](https://tech-blog.rakus.co.jp/entry/20200826/docker)
2. [uid、gidの問題](https://stackoverflow.com/questions/52706424/how-can-docker-container-write-to-a-mounted-directory-with-permissions-granted-t)
でまだ調べております。

Reply:
`pwd`？
`ls -la`？

Reply:
`yokoteru@yokosemominoMBP paddle % ls -la
total 120
drwxr-xr-x  17 yokoteru  staff    544  4 12 01:00 .
drwxr-xr-x   6 yokoteru  staff    192  4 11 23:40 ..
drwxr-xr-x   3 yokoteru  staff     96  4 12 00:11 .cache
drwxr-xr-x  12 yokoteru  staff    384  4 11 23:40 .git
-rw-r--r--   1 yokoteru  staff     35  4 11 23:40 .gitignore
-rw-r--r--   1 yokoteru  staff    704  4 11 23:40 Dockerfile
-rw-r--r--   1 yokoteru  staff    467  4 11 23:40 Makefile
-rw-r--r--   1 yokoteru  staff   1828  4 11 23:40 README.md
drwxr-xr-x   3 yokoteru  staff     96  4 11 23:40 db
-rw-r--r--   1 yokoteru  staff    875  4 11 23:40 docker-compose.yml
-rw-r--r--   1 yokoteru  staff    640  4 11 23:40 go.mod
-rw-r--r--   1 yokoteru  staff  26715  4 11 23:40 go.sum
drwxr-xr-x   6 yokoteru  staff    192  4 11 23:40 internal
-rw-r--r--   1 yokoteru  staff    736  4 11 23:40 main.go
drwxr-xr-x   4 yokoteru  staff    128  4 11 23:40 pkg
-rw-r--r--   1 yokoteru  staff    166  4 11 23:40 sqlboiler.toml
drwxrwxrwx  12 yokoteru  staff    384  4 12 01:01 web`

Reply:
```
yokoteru@yokosemominoMBP paddle % cd web
yokoteru@yokosemominoMBP web % ls -la
total 2184
drwxrwxrwx  12 yokoteru  staff     384  4 12 01:01 .
drwxr-xr-x  17 yokoteru  staff     544  4 12 01:00 ..
drwxr-xr-x   2 yokoteru  staff      64  4 12 01:01 .cache
-rw-r--r--   1 yokoteru  staff     310  4 11 23:40 .gitignore
-rw-r--r--   1 yokoteru  staff     116  4 11 23:40 .yarnrc
drwxrwxrwx  22 yokoteru  staff     704  4 12 01:03 node_modules
-rwxrwxrwx   1 yokoteru  staff     976  4 11 23:40 package.json
drwxrwxrwx   8 yokoteru  staff     256  4 11 23:40 public
drwxrwxrwx  11 yokoteru  staff     352  4 11 23:40 src
-rwxrwxrwx   1 yokoteru  staff     535  4 11 23:40 tsconfig.json
-rwxrwxrwx   1 yokoteru  staff  533809  4 12 01:10 yarn-error.log
-rwxrwxrwx   1 yokoteru  staff  509485  4 11 23:40 yarn.lock
```

Reply:
ゴリ押してweb配下の権限を変更していくとログも変化しているので、解決できそうな感じがします。
`error An unexpected error occurred: "EACCES: permission denied, mkdir '/home/app/node_modules/@pmmmwh/react-refresh-webpack-plugin/node_modules'".`

Reply:
```
rm -r ./web/node_modules
docker-compose up
```

Reply:
node_modules配下削除したところ、別箇所で修正必要のようでそちら直します。
T```
raceback (most recent call last):
  File "docker-compose", line 3, in <module>
  File "compose/cli/main.py", line 81, in main
  File "compose/cli/main.py", line 199, in perform_command
  File "compose/cli/command.py", line 60, in project_from_options
  File "compose/cli/command.py", line 152, in get_project
  File "compose/cli/docker_client.py", line 41, in get_client
  File "compose/cli/docker_client.py", line 170, in docker_client
  File "docker/api/client.py", line 197, in __init__
  File "docker/api/client.py", line 221, in _retrieve_server_version
```

Reply:
お騒がせしました、web側の立ち上げに関するエラーについて解消しました 🙇🏻  

【解決法】
・nodemodules配下のファイル削除(不要かもですが)
・Virtualbox->docker for Mac　

![image](https://user-images.githubusercontent.com/21329889/114314433-8f056700-9b35-11eb-8c5f-aceda93fb4ff.png)